datanode/sqlstore/accounts.go

```diff
@@ -114,10 +100,15 @@ func (as *Accounts) GetAll(ctx context.Context) ([]entities.Account, error) {
 func (as *Accounts) Obtain(ctx context.Context, a *entities.Account) error {
 	as.cacheLock.Lock()
 	defer as.cacheLock.Unlock()
-	if account, ok := as.checkCache(a.PartyID, a.AssetID, a.MarketID, a.Type); ok {
+
+	id := deterministicIDFromAccount(a)
+	if account, ok := as.idToAccount[id]; ok {
 		a.ID = account.ID
+		a.VegaTime = account.VegaTime
```

Comment:
@pscott31 I set these attrs as well, any reason this is not the right thing to do?